wagtail/permission_policies/pages.py

```diff
@@ -152,13 +153,18 @@ def instances_with_direct_explore_permission(self, user):
             ]
 
     def explorable_root_instance(self, user):
+        # This method is used all around the admin via get_explorable_root_page,
+        # so cache the result on the user for the duration of the request
+        if hasattr(user, self._explorable_root_instance_cache_name):
+            return getattr(user, self._explorable_root_instance_cache_name)
```

Comment:
I don't like doing this, but the logic in this method (previously in `get_explorable_root_page` in `navigation.py`) is used in many other places in the admin.

We need some place to cache this, and moving this to `UserPagePermissionsProxy` wouldn't help because the code that uses this method doesn't always have access to the template context.